### PR TITLE
Spotify API with Spotipy Part 1

### PR DESCRIPTION
**Pull request by joshjwelsh**
This article is the first in a 3 part series that teaches the user how to use the spotipy api. It is made for beginners. This is my first upload so wish me luck!

### CODE REVIEW

**Comment by hectorkambow** [comment]
@joshjwelsh 
Also FYI - be sure to include the # of the issue (approved topic form) to be able to link it to your PR. 
This allows the linked issue to be closed once the PR is successfully merged 🚀

**Comment by joshjwelsh** [comment]
@hectorkambow thank you! Will add it to my main pull request next time for now ill just add it here. 
#907

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+![img](py_img.jpg)
+
+# Spotify API with Spotipy Part 1
+
+## Introduction 
+
+Spotify is a wonderful music streaming app with powerful API functionality. The API is both beginner-friendly and offers a wide range of endpoints. This makes Spotify API a great way to get started for beginner programmers who are interested in getting involved in web APIs and even data science, or advanced programmers who want more control over their playlists. Spotipy is a popular wrapper for the Spotify API. It enables and supports all the features of the Spotify API but allows them to be designed and run in python. This allows for scalability and also will come in handy when we start dealing with large amounts of data.
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md]: Add a link to Spotipy's website when you first mention Spotipy.

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Spotify is a wonderful music streaming app with powerful API functionality. The API is both beginner-friendly and offers a wide range of endpoints. This makes Spotify API a great way to get started for beginner programmers who are interested in getting involved in web APIs and even data science or advanced programmers who want more control over their playlists. Spotipy is a popular wrapper for the Spotify API. It enables and supports all the features of the Spotify API but allows them to be designed and run in Python. This allows for scalability and also will come in handy when we start dealing with large amounts of data.
+
+## Goals 
+
+This is part one of a three-part tutorial series on Spotipy and its applications. By the end, you should be able to design and implement algorithms for creating unique playlists, search for recommended music, and analyze metadata about your music selection.
+
+To achieve these lofty goals will require a strong foundation. That's the goal of part 1 of this series. By the end, you should understand how to set up a spotipy project, the authentification process, and finishing it off. We'll start using some of the spotipy endpoints.
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:13]: authentication process

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+## Prerequisites
+
+1. You need a Spotify premium account
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:17]: Are there any alternatives? I don't think everyone has a premium account.

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+2. I try to make my tutorials as simple as possible, so even if you are a beginner programmer, you can follow along and achieve something you're proud of. That being said, you should have some basic knowledge of Python before starting. If you know how to use the [enumerate()](https://docs.python.org/3/library/functions.html#enumerate) function, how to [import modules](https://docs.python.org/3/tutorial/modules.html), and how to use pip3, you should be able to complete this part of the tutorial. If you are not confident in these skills, yet it is okay! Click on the links and when you feel confident, return to the article.
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:19]: add a link to pip3 as well.

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+3. You should understand how to use the terminal
+
+
+## Step 1: Set up a Developer account
+
+
+The first step is to create a [Spotify developer account](https://developer.spotify.com/dashboard/). Click on the link, sign in to your Spotify account, and follow the instructions to create a developer account.
+For the redirect URL, it is highly recommended to use http://localhost:9000 or the localhost port of your choosing.
+
+Trust me, I have tried using a different website, and it usually creates bugs. Localhost is the best option for the redirect link.
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:30]: Don't use trust me, I speak with experience, etc. It comes across as way too informal.

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+2. I try to make my tutorials as simple as possible, so even if you are a beginner programmer, you can follow along and achieve something you're proud of. That being said, you should have some basic knowledge of Python before starting. If you know how to use the [enumerate()](https://docs.python.org/3/library/functions.html#enumerate) function, how to [import modules](https://docs.python.org/3/tutorial/modules.html), and how to use pip3, you should be able to complete this part of the tutorial. If you are not confident in these skills, yet it is okay! Click on the links and when you feel confident, return to the article.
+
+3. You should understand how to use the terminal
+
+
+## Step 1: Set up a Developer account
+
+
+The first step is to create a [Spotify developer account](https://developer.spotify.com/dashboard/). Click on the link, sign in to your Spotify account, and follow the instructions to create a developer account.
+For the redirect URL, it is highly recommended to use http://localhost:9000 or the localhost port of your choosing.
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:28]: image

Well... I created an account and landed here. Your article does not talk about what to do next!

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+Trust me, I have tried using a different website, and it usually creates bugs. Localhost is the best option for the redirect link.
+
+You will need to save your client ID and client secret for the authentication process.
+
+
+## Step 2: Creating a python environment 
+
+
+Any good developer has an organized file system. If this is your first python project, consider making a directory called PythonScripts (you can name it whatever you want). Inside that directory, create a folder called spotipy_tutorial.
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:38]: Don't put it across as any good developer has an organized file system. Instead, A good developer's practice is to have an organized file system.

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+Now run the following command to recreate a virtual environment. **Do not skip this step**. If you already have virtualenv installed, run the line below:
+
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:41]: Use code formatting.

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    ~/PythonScripts/spotipy_tutorial >> virtualenv venv
+
+If you have never installed a virtual environment before or the line above doesn't work, do the following: 
+
+    ~/PythonScripts/spotipy_tutorial >> python3 -m pip3 install --user virtualenv
+    ~/PythonScripts/spotipy_tutorial >> virtualenv venv
+
+
+Why create a venv?
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:50]: ### Why create a venv?

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+If you have never installed a virtual environment before or the line above doesn't work, do the following: 
+
+    ~/PythonScripts/spotipy_tutorial >> python3 -m pip3 install --user virtualenv
+    ~/PythonScripts/spotipy_tutorial >> virtualenv venv
+
+
+Why create a venv? 
+
+Creating a virtual environment is usually overlooked by beginner programmers, but this is a bad mistake to make. New versions of Python are released almost every year, and this can cause problems when you want to return to a project. Modules start to become incompatible with the new versions of Python, and your code begins to break. Trust me, I speak from experience; it is much harder to fix your program after it breaks (especially after 6+ months of not reading your code). Additionally, some modules can have bugs that can do some nasty stuff to your system. By creating a self-contained directory, you can avoid stress and worry.
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:52]: Incorporate suggestions given earlier.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
How to obtain the Spotify credentials? Include a section on that! Show the process of obtaining the credentials (secret keys).

File: articles/spotify-python-part-1/spotipy_spotify_python.md (hunk 8 above)
Comment (joshjwelsh) [reply articles/spotify-python-part-1/spotipy_spotify_python.md:41]: Could you explain? there is no code here and I use the code formatting on the following lines.

Comment (lalith1403) [reply articles/spotify-python-part-1/spotipy_spotify_python.md:41]: Sorry, The next line requires formatting.

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+
+Any good developer has an organized file system. If this is your first python project, consider making a directory called PythonScripts (you can name it whatever you want). Inside that directory, create a folder called spotipy_tutorial.
+
+Now run the following command to recreate a virtual environment. **Do not skip this step**. If you already have virtualenv installed, run the line below:
+
+    ~/PythonScripts/spotipy_tutorial >> virtualenv venv
+
+If you have never installed a virtual environment before or the line above doesn't work, do the following: 
+
+    ~/PythonScripts/spotipy_tutorial >> python3 -m pip3 install --user virtualenv
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:46]: code formatting. 

```bash
 ~/PythonScripts/spotipy_tutorial >> python3 -m pip3 install --user virtualenv
```

File: articles/spotify-python-part-1/spotipy_spotify_python.md (hunk 2 above)
Comment (joshjwelsh) [reply articles/spotify-python-part-1/spotipy_spotify_python.md]: fixed

File: articles/spotify-python-part-1/spotipy_spotify_python.md (hunk 1 above)
Comment (joshjwelsh) [reply articles/spotify-python-part-1/spotipy_spotify_python.md]: fixed

File: articles/spotify-python-part-1/spotipy_spotify_python.md (hunk 4 above)
Comment (joshjwelsh) [reply articles/spotify-python-part-1/spotipy_spotify_python.md]: fixed

File: articles/spotify-python-part-1/spotipy_spotify_python.md (hunk 6 above)
Comment (joshjwelsh) [reply articles/spotify-python-part-1/spotipy_spotify_python.md:30]: I appended the set up developer account section so it now has the steps to get your client id. This is fixed.

File: articles/spotify-python-part-1/spotipy_spotify_python.md (hunk 11 above)
Comment (joshjwelsh) [reply articles/spotify-python-part-1/spotipy_spotify_python.md]: fixed

File: articles/spotify-python-part-1/spotipy_spotify_python.md (hunk 9 above)
Comment (joshjwelsh) [reply articles/spotify-python-part-1/spotipy_spotify_python.md]: I made it bold.

**Comment by joshjwelsh** [comment]
@lalith1403 I hope its good to go now. I fixed all the suggestions. Thank you!

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+`export SPOTIPY_CLIENT_ID='your-spotify-client-id'`
+
+`export SPOTIPY_CLIENT_SECRET='your-spotify-client-secret'`   
+
+`export SPOTIPY_REDIRECT_URI='your-spotify-redirect-url'`
+
+I recommend you create a file called cred.py and store your values inside it. It goes without saying that this file should not be committed to a repository. 
+
+Create a cred.py file and put the following in your file: 
+
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:86]: All the code blocks need to be formatted properly. For example, 
```py
    client_ID='your-spotify-client-id'
    client_SECRET='your-spotify-client-secret'   
    redirect_url='your-spotify-redirect-url'
```

File: articles/spotify-python-part-1/spotipy_spotify_python.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+    client_ID='your-spotify-client-id'
+    client_SECRET='your-spotify-client-secret'   
+    redirect_url='your-spotify-redirect-url'
+
+Now create a file main.py and add the following boilerplate code: 
+
+    import spotipy
```
Comment (lalith1403) [inline articles/spotify-python-part-1/spotipy_spotify_python.md:93]:
```py
    import spotipy
    from spotipy.oauth2 import SpotifyOAuth
    import cred 
```

Comment (lalith1403) [reply articles/spotify-python-part-1/spotipy_spotify_python.md:93]: Refer to this [article ](https://www.freecodecamp.org/news/how-to-format-code-in-markdown/#)for more details.

**Review by lalith1403 — CHANGES_REQUESTED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@joshjwelsh 
Going over your article - almost ready to publish - but it not quite ready - as we will need an author page (short bio) and headshot image to include towards the bottom of all your articles.

Please see any example in the authors folder in the repo thank you 👍

**Comment by hectorkambow** [comment]
author pinged via slack

**Comment by hectorkambow** [comment]
image

**Comment by hectorkambow** [comment]
@joshjwelsh  

Since this is your 1st article you are contributing - please be sure to contribute a short author bio and headshot image. 

Please refer to our author folder in the repo for examples 🚀

**Pull request merged**